src/select_any.rs

```diff
@@ -0,0 +1,77 @@
+use std::mem;
+use std::prelude::v1::*;
+
+use {Future, IntoFuture, Poll, Async};
+
+/// Future for the `select_any` combinator, waiting for one of any of a list of
+/// futures to succesfully complete. unlike `select_all`, this future ignores all
+/// but the last error, if there are any.
+///
+/// This is created by this `select_any` function.
+#[must_use = "futures do nothing unless polled"]
+pub struct SelectAny<A> where A: Future {
+    inner: Vec<A>,
+}
+
+/// Creates a new future which will select the first successful future over a list of futures.
+///
+/// The returned future will wait for any future within `list` to be ready and Ok. Unlike
+/// select_all, this will only return the first successful completion, or the last
+/// failure. This is useful in contexts where any success is desired and failures
+/// are ignored, unless all the futures fail.
+///
+/// # Panics
+///
+/// This function will panic if the iterator specified contains no items.
+pub fn select_any<I>(iter: I) -> SelectAny<<I::Item as IntoFuture>::Future>
+    where I: IntoIterator,
+          I::Item: IntoFuture,
+{
+  let ret = SelectAny {
+    inner: iter.into_iter()
+    .map(|a| a.into_future())
+    .collect(),
+  };
+  assert!(ret.inner.len() > 0);
+  ret
+}
+
+impl<A> Future for SelectAny<A> where A: Future {
+  type Item = (A::Item, Vec<A>);
```

Comment:
Ah I personally avoid this in repos for now, but when rustfmt comes along hopefully that'll help!